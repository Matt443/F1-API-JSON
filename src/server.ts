export { getDriverLineups } from "./scraper/driver-lineup";
export { getTeamLineups } from "./scraper/team-lineup";
export { getDriverStandings } from "./scraper/driver-standings";
export { getConstructorStandings } from "./scraper/constructors-standings";
export { getWorldChampions } from "./scraper/world-champions";
export { getRaceResults } from "./scraper/race-results";
