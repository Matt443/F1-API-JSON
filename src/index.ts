import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
