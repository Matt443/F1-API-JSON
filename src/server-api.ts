import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
