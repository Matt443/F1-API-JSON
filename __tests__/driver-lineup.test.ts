import { getDriverLineups } from "../src/server";

test("current driver lineup", async () => {
    expect(await getDriverLineups()).toMatchSnapshot();
});
