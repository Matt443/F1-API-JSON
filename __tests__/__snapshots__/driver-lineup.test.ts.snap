// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`current driver lineup 1`] = `
[
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/M/MAXVER01_Max_Verstappen/maxver01.png.transform/2col/image.png",
    "name": "Max Verstappen",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Netherlands.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/MAXVER01.png.transform/2col/image.png",
    "team": "Red Bull Racing",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/S/SERPER01_Sergio_Perez/serper01.png.transform/2col/image.png",
    "name": "Sergio Perez",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Mexico.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/SERPER01.png.transform/2col/image.png",
    "team": "Red Bull Racing",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/C/CHALEC01_Charles_Leclerc/chalec01.png.transform/2col/image.png",
    "name": "Charles Leclerc",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Monaco.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/CHALEC01.png.transform/2col/image.png",
    "team": "Ferrari",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/C/CARSAI01_Carlos_Sainz/carsai01.png.transform/2col/image.png",
    "name": "Carlos Sainz",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Spain.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/CARSAI01.png.transform/2col/image.png",
    "team": "Ferrari",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/G/GEORUS01_George_Russell/georus01.png.transform/2col/image.png",
    "name": "George Russell",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/United%20Kingdom.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/GEORUS01.png.transform/2col/image.png",
    "team": "Mercedes",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/L/LEWHAM01_Lewis_Hamilton/lewham01.png.transform/2col/image.png",
    "name": "Lewis Hamilton",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/United%20Kingdom.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/LEWHAM01.png.transform/2col/image.png",
    "team": "Mercedes",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/E/ESTOCO01_Esteban_Ocon/estoco01.png.transform/2col/image.png",
    "name": "Esteban Ocon",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/France.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/ESTOCO01.png.transform/2col/image.png",
    "team": "Alpine",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/P/PIEGAS01_Pierre_Gasly/piegas01.png.transform/2col/image.png",
    "name": "Pierre Gasly",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/France.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/PIEGAS01.png.transform/2col/image.png",
    "team": "Alpine",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/O/OSCPIA01_Oscar_Piastri/oscpia01.png.transform/2col/image.png",
    "name": "Oscar Piastri",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Australia.jpg.transform/2col/image.jpg",
    "rank": undefined,
    "team": "McLaren",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/L/LANNOR01_Lando_Norris/lannor01.png.transform/2col/image.png",
    "name": "Lando Norris",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/United%20Kingdom.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/LANNOR01.png.transform/2col/image.png",
    "team": "McLaren",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/V/VALBOT01_Valtteri_Bottas/valbot01.png.transform/2col/image.png",
    "name": "Valtteri Bottas",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Finland.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/VALBOT01.png.transform/2col/image.png",
    "team": "Alfa Romeo",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/G/GUAZHO01_Guanyu_Zhou/guazho01.png.transform/2col/image.png",
    "name": "Guanyu Zhou",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/China.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/GUAZHO01.png.transform/2col/image.png",
    "team": "Alfa Romeo",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/L/LANSTR01_Lance_Stroll/lanstr01.png.transform/2col/image.png",
    "name": "Lance Stroll",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Canada.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/LANSTR01.png.transform/2col/image.png",
    "team": "Aston Martin",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/F/FERALO01_Fernando_Alonso/feralo01.png.transform/2col/image.png",
    "name": "Fernando Alonso",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Spain.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/FERALO01.png.transform/2col/image.png",
    "team": "Aston Martin",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/K/KEVMAG01_Kevin_Magnussen/kevmag01.png.transform/2col/image.png",
    "name": "Kevin Magnussen",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Denmark.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/KEVMAG01.png.transform/2col/image.png",
    "team": "Haas F1 Team",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/N/NICHUL01_Nico_Hulkenberg/nichul01.png.transform/2col/image.png",
    "name": "Nico Hulkenberg",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Germany.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/NICHUL01.png.transform/2col/image.png",
    "team": "Haas F1 Team",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/N/NYCDEV01_Nyck_De%20Vries/nycdev01.png.transform/2col/image.png",
    "name": "Nyck De Vries",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Netherlands.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/NYCDEV01.png.transform/2col/image.png",
    "team": "AlphaTauri",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/Y/YUKTSU01_Yuki_Tsunoda/yuktsu01.png.transform/2col/image.png",
    "name": "Yuki Tsunoda",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Japan.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/YUKTSU01.png.transform/2col/image.png",
    "team": "AlphaTauri",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/A/ALEALB01_Alexander_Albon/alealb01.png.transform/2col/image.png",
    "name": "Alexander Albon",
    "nationalityImage": "https://www.formula1.com/content/dam/fom-website/flags/Thailand.jpg.transform/2col/image.jpg",
    "rank": "https://www.formula1.com/content/dam/fom-website/2018-redesign-assets/drivers/number-logos/ALEALB01.png.transform/2col/image.png",
    "team": "Williams",
  },
  {
    "driverImage": "https://www.formula1.com/content/dam/fom-website/drivers/L/LOGSAR01_Logan_Sargeant/logsar01.png.transform/2col/image.png",
    "name": "Logan Sargeant",
    "nationalityImage": undefined,
    "rank": undefined,
    "team": "Williams",
  },
]
`;
